types/validator_set.go

```diff
@@ -953,3 +963,33 @@ func safeSubClip(a, b int64) int64 {
 	}
 	return c
 }
+
+func safeMul(a, b int64) (int64, bool) {
```

Comment:
Would it not be more efficient to just check that `|a| < MaxInt/ |b|` like for safeAdd?